README.md

```diff
@@ -86,6 +86,34 @@ Sidekiq::Ultimate::Resurrector::Count.read(:job_id => "2647c4fe13acc692326bd4c2"
 => 1
 ```
 
+### Empty queues list refresh interval
+
+```ruby
+Sidekiq::Ultimate.setup! do |config|
+  config.empty_queues_refresh_interval_sec = 42
+end
+```
+
+Specifies how often the list of empty queues should be refreshed.
+In a nutshell, this sets the maximum possible delay between when a job was pushed to previously empty queue and earliest the moment when that new job could be picked up.
+
+**Note:** every worker maintains its own local list of empty queues.
+Setting this interval to a low value will increase the number of Redis calls needed to check for empty queues, increasing the total load on Redis.
+
+This setting helps manage the tradeoff between performance penalties and latency needed for reliable fetch.
+Under the hood, Sidekiq's default fetch occurs with [a single Redis `BRPOP` call](https://redis.io/commands/brpop/) which is passes list of all queues to pluck work from.
+In contrast, [reliable fetch uses `LPOPRPUSH`](https://redis.io/commands/rpoplpush/) (or the equivalent `LMOVE` in later Redis versions) to place in progress work into a WIP queue.
+However, `LPOPRPUSH` can only check one source queue to pop from at once, and [no multi-key alternative is available](https://github.com/redis/redis/issues/1785), so multiple Redis calls are needed to pluck work if an empty queue is checked.
+In order to avoid performance penalties for repeated calls to empty queues, Sidekiq Ultimate therefore maintains a list of recently know empty queues which it will avoid polling for work.
+
+Therefore:
+- If your Sidekiq architecture has *a low number of total queues*, the worst case penalty for polling empty queues will be bounded, and it is reasonable to **set a shorter refresh period**.
+- If your Sidekiq architecture has a *high number of total queues*, the worst case penalty for polling empty queues is large, and it is recommended to **set a longer refresh period**.
+- When adjusting this setting:
+    - Check that work is consumed appropriately quickly from high priority queues after they bottom out (after increasing the refresh interval)
+    - Check that backlog work does not accumulate in low priority queues (after decreasing the refresh interval)
```

Comment:
I really like that you added all this detail but in order to make it a bit digestible, please rewrite it as suggested below:

- Start by explaining the problem this feature solves
- Explain how it works under the hood (the different types of Redis operations like you do now)
- Explain what the performance implications are for a shared global queue (also please include the total number of workers into the description, it looks like its missing)
- Then explain the constant itself

Reply:
@davidbodow-st any thoughts since you proposed this section?

Reply:
@stefanl-st @sensorsasha TBH, I just wrote this off the top of my head, so we can sink more time into refining the structure and wording if needed, though I think it's probably at diminishing returns right now.

I also disagree with the proposed structure, as IMHO, it would provide too much detail for quick reference use cases, which are the majority of documentation readers' time. Here, I assume that most readers will need to learn the "why" just once and the "what was that config option called again?" many times while using the feature.

So, my preferred structure is:
- Quick reference overview
- Explain the problem that needs to be solved
- Explain the tradeoff that the implementation details require us to make

IMO we're already following this structure, and it's mostly optimized for the quick reference use case. If we want to take another pass, then I would recommend:
- Tighten up the "Therefore" section and move it into the quick reference, for readers who care about making the correct decision without fully building the mental model of the implementation that they need to do that
- Take a general pass to make the language more concise (could definitely be condensed)

@stefanl-st If you disagree / would still prefer your structure, can you elaborate on which audience you want to optimize for instead of quick reference readers (who would expect the `Then explain the constant itself` part at the very beginning, IMO)?